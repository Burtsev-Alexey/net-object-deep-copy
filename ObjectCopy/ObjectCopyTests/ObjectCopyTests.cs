using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using Newtonsoft.Json;
using NUnit.Framework;
using ObjectCopy;

namespace ObjectCopyTests
{
    public class ObjectCopyTests
    {
        private TestObject originalObject;
        private TestObject clonedObject;

        [SetUp]
        public void SetupTest()
        {
            this.originalObject = new TestObject();
            this.clonedObject = this.originalObject.Clone();
        }

        [TearDown]
        public void TearDownTest()
        {
            this.originalObject.Dispose();
            this.clonedObject.Dispose();
        }

        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void TestObjectDeepCopyToBaseline() //Ensure all required fields, collections items etc
        {
            //Ensure all field have been copied by using an approved baseline
            //This is useful in testing breaking changes
            Approvals.Verify(JsonConvert.SerializeObject(this.clonedObject));
        }

        [Test]
        public void TestObjectDeepCopyEnsureIsDeepCopy()//Verify that all objects are deep copied
        {
            var cloneObjectsAreUnique = this.originalObject.Verify(this.clonedObject);

            //this.clonedObject.DictionaryProp["A"] = this.originalObject.DictionaryProp["A"];//force ensure that collections are checked
            var cloneObjectsAreUnique2 = this.originalObject.Verify(this.clonedObject);
            var sameObjectNotUnique = this.originalObject.Verify(this.originalObject);
        }
    }
}
