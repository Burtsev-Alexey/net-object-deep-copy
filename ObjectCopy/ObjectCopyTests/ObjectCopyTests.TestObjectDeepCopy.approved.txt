{"StringProp":"One","ObjectProp":{"StringProp":"Three"},"ObjectCollection":[{"StringProp":"Three"},{"StringProp":"Three"},{"StringProp":"Three"}],"ObjectCollection2":[{"StringProp":"Three"},{"StringProp":"Three"},{"StringProp":"Three"},{"StringProp":"Three"}],"DictionaryProp":{"A":{"StringProp":"Three"},"B":{"StringProp":"Three"},"C":{"StringProp":"Three"},"D":{"StringProp":"Three"}}}